php/libraries/Database.class.inc

```diff
@@ -635,6 +635,13 @@ class Database
         }
 
         $rows = $prepared->fetchAll(PDO::FETCH_ASSOC);
+        // The fetchAll function will return an empty set if no results are
+        // found and FALSE on "failure". We don't expect this case so an
+        // exception is thrown.
+        if (is_bool($rows)) {
```

Comment:
this isn't checking if it's false, it's checking if it's a boolean. It should be `=== false`

Reply:
I tried that at first but `phan` gave me an error about comparing an array to a boolean (in the case where it was NOT false).

I'll change this statement to include both and see how phan feels about it.

Reply:
the docs pretty clearly say that the return type is `array|false`, so if phan is complaining about checking the false case it sounds like a bug in phan.

Reply:
How about `!is_array()` ?